mlrun/datastore/google_cloud_storage.py

```diff
@@ -84,7 +51,30 @@ def get_filesystem(self, silent=True):
         return self._filesystem
 
     def get_storage_options(self):
-        return dict(token=self._get_secret_or_env("GOOGLE_APPLICATION_CREDENTIALS"))
+        google_credentials = self._get_secret_or_env("GOOGLE_APPLICATION_CREDENTIALS")
+        gcp_credentials = self._get_secret_or_env("GCP_CREDENTIALS")
+        if (
+            gcp_credentials
+            and google_credentials
+            and google_credentials != gcp_credentials
```

Comment:
In `google_credentials` you Will get a path to a credential file.
They should never be the same!

If you want, you can read the file and compare its data.